tensorflow_probability/python/bijectors/chain.py

```diff
@@ -241,7 +241,7 @@ def _is_increasing(self, **kwargs):
 
   def _inverse(self, y, **kwargs):
     for b in self.bijectors:
-      y = b.inverse(y, **kwargs.get(b.name, {}))
+      y = b.inverse(y, **kwargs.get(b.name, {}), **kwargs)
```

Comment:
The problem was that we want to use the chain within a transformed distribution so we would need `chain.forward()`, `chain.inverse()` and `chain.log_det_jacobian()` available, and since they are wrapped in transformed distribution then we don't have the ability to pass the forward, inverse and jacobian. I might be missing something of course and there could be an easier way to do this? Otherwise we could always do 

`y = b.inverse(y, **kwargs.get(b.name, {}), **kwargs.get("conditional", {}))`

For some reason, at the time, I felt this was less elegant. I was actually passing lots of things to the bijector before settling on just using the conditional parameters though, so perhaps that's why.

Reply:
Although, `**kwargs.get("conditional", {})` wouldn't work because kwargs["conditional"] is a  tensor. Yeah, it's not clear to me how to pass the conditional argument to one or many named bijectors in the chain when the chain is passed to a transformed distribution.

Reply:
I've got this working in my code, but it does involve adding extra lambda functions which implement these steps into the transformed distribution
`
put_conditional = lambda conditional : dict(
        zip(["MADE_{}".format(i) for i in range(n_mades)], 
            [{"conditional": conditional} for i in range(n_mades)]))
distribution.log_prob = lambda a, conditional=None : distribution.log_prob(a, bijector_kwargs=put_conditional(conditional)
`
One of the problems I have is that I have a bunch of different distributions, some with bijectors, some without, some are chains, some are variational networks, etc. Each of these takes a different argument in log_prob, sample, etc. This means I can't write code like
`any_tfp_object.log_prob(input, dict("conditional"=conditional_input))`
and get it to work. Instead I have to know, beforehand, what the object is and whether it takes `bijector_kwargs` or `kwargs` or is self defined and takes `conditional` directly. It would be nice to have something that was much more general. What do you think?

Reply:
Yeah, it's definitely far from ideal. I'll bring this up internally, and we'll see if people have more ideas. I think apriori its not palatable to require leaf bijectors be tolerant of extraneous kwargs. Perhaps there's a way to do it in a way that only knows the names of those bijectors, and not their structure.

For now, I can only suggest keeping structure the same. I.e. always use a transformed distribution and always use chain, and specialize by having a chain of a single bijector when you want a single bijector, or a chain of `tfb.Identity` when you don't want a transformed (in the mathematical sense) distribution. Then, your lambda should be the same for all conditions, just differing in the number of chain elements (and whether they need to be conditioned).

Reply:
Great thanks. I'd be super interested in the outcome ;)